### PR TITLE
Add the setup script to install anything you need to run the work demo

### PR DESCRIPTION
**Pull request by Leo6Leo**

### CODE REVIEW

File: hack/setup.sh [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+kubectl apply -f https://github.com/knative/net-kourier/releases/download/knative-v1.12.3/kourier.yaml
+
+echo "Configuring Knative Serving to use Kourier..."
+kubectl patch configmap/config-network \
+  --namespace knative-serving \
+  --type merge \
+  --patch '{"data":{"ingress-class":"kourier.ingress.networking.knative.dev"}}'
+
+echo "Fetching External IP or CNAME..."
+kubectl --namespace kourier-system get service kourier
```
Comment (Cali0707) [inline hack/setup.sh]: I don't think we actually need this, this is just verifying that kourier is running, but we don't check it anywhere. Maybe we can remove it for now?

File: hack/setup.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        sleep 5
+        kubectl get pods --namespace knative-serving | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods --namespace knative-serving | grep -v NAME)
+
+echo "All pods are running."
+
+echo "Verifying Knative Serving Installation..."
+kubectl get pods --namespace knative-serving
+
```
Comment (Cali0707) [inline hack/setup.sh:37]: What's the point of this if we had the previous loop check that all pods are running?

File: hack/setup.sh [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    GO111MODULE=on go get github.com/google/ko/cmd/ko
+
+    # Verifying ko installation
+    if command -v ko >/dev/null 2>&1; then
+        echo "ko successfully installed. Proceeding with the installation of the config"
+    else
+        echo "Failed to install ko. Please try to manually install ko."
+        exit 1
+    fi
+fi
```
Comment (Cali0707) [inline hack/setup.sh]: I don't think installing ko should be a part of this script. We should just let users know to install ko themselves

File: hack/setup.sh [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    if [ "$POD_STATUS" != "Running" ] && [ "$POD_STATUS" != "Completed" ]; then
+        echo "Waiting for pod $POD_NAME to reach Running or Completed state..."
+        sleep 5
+        kubectl get pods -n knative-eventing | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods -n knative-eventing | grep -v NAME)
+
+echo "All the pods are in the desired state."
+
+echo "You are good to go! Everything you need have been installed successfully."
```
Comment (Cali0707) [inline hack/setup.sh]:
```suggestion

```
Seems like a little extra printing, not needed

File: hack/setup.sh [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+while IFS= read -r line; do
+    POD_NAME=$(echo $line | awk '{print $1}')
+    POD_STATUS=$(echo $line | awk '{print $3}')
+    if [ "$POD_STATUS" != "Running" ] && [ "$POD_STATUS" != "Completed" ]; then
+        echo "Waiting for pod $POD_NAME to reach Running or Completed state..."
+        sleep 5
+        kubectl get pods -n knative-eventing | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods -n knative-eventing | grep -v NAME)
```
Comment (Cali0707) [inline hack/setup.sh]:
```suggestion
done < <(kubectl get pods -n knative-samples | grep -v NAME)
```
The example is in a different namespace :)

File: hack/setup.sh (hunk 2 above)
Comment (Leo6Leo) [reply hack/setup.sh:37]: This is just print the complete list of the running pods in the knative-serving namespace, as in other knative script we do this at the end of each section.

File: hack/setup.sh (hunk 3 above)
Comment (Leo6Leo) [reply hack/setup.sh]: Sounds good. Instead I will prompt user that ko is not installed, and ask them to install manually.

**Review by Cali0707 — APPROVED** [review]
Thanks for fixing this @Leo6Leo !

**Pull request merged**